diagnostic.md

```diff
@@ -6,72 +6,87 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```bash
-// your response here
+The puropse of a backend is to save data. It allows data to persist when browser is closed.
+In addition, it allows two users to interact with each other in the app
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```bash
-// your response here
+model fetches data in the database
 ```
 
 Which layer in the MVC pattern communicates with the model?
 
 ```bash
-// your response here
+controller(lazy- most often skinny- manager) communicatese with the model
 ```
 
 Why don't we use views in our interpretation of the MVC pattern?
 
 ```bash
-// your response here
+We do not need views . We are using our own view layer. My Question(when there it is a single page app, is the view neccessary?).
 ```
 
 What does C.R.U.D stand for?
 
 ```bash
-// your response here
+Create read update destroy
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
 ```bash
-// your response here
+The controller
 ```
 
 A user action fires a `GET` request for `person/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```bash
-// your response here
+-browser fires a request to the webserver,
+-the webserve goes to the router.
+-route indicate to the server which controllers should be triggered (and how) by which kinds of requests
+- then the controller ask model to do a 'GET' request for 'person/1'
+-Model talks to database and make the 'GET' request and retrieves the data from the database
+- Model then gives the data to the controller.
+-  Controller than gives it to view to display.
 ```
```

Comment:
if you're using a view layer, the view gets passed back to the controller then the controller delivers that to the server, but don't worry about views for now.